packages/oauth/oauth-provider/src/oauth-provider.ts

```diff
@@ -1215,20 +1254,14 @@ export class OAuthProvider extends OAuthVerifier {
           csrfCookie(input.request_uri),
         )
 
-        const { deviceId } = await deviceManager.load(req, res)
+        const { deviceId } = await deviceManager.load(req, res, true)
 
-        const { account, info } = await server.signIn(
+        return server.signIn(
           deviceId,
+          input.request_uri,
+          input.client_id,
           input.credentials,
         )
-
-        // Prevent fixation attacks
-        await deviceManager.rotate(req, res, deviceId)
```

Comment:
Is this mitigation no longer necessary here?

Reply:
This is covered by the `true` argument on the `deviceManager.load()` call.

https://github.com/bluesky-social/atproto/pull/2734/files/24e0f45ab7c957861072e639b6ecb7b64a4d7444#diff-c8e565c82e08b3dec17e4b2c34b802572ec1f20822160e72891d3c60178f3371R1257